cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1095 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"golang.org/x/net/html"
+)
+
+const (
+	githubPath        = "https://api.github.com/repos/%s/commits/master"
+	jsonFile          = "updates.json"
+	github            = "GitHub Sha Check"
+	htmlScrape        = "HTML String Check"
+	pathOkCoin        = "https://www.okcoin.com/docs/en/#change-change"
+	pathOkex          = "https://www.okex.com/docs/en/#change-change"
+	pathBTSE          = "https://www.btse.com/apiexplorer/spot/#btse-spot-api"
+	pathBitfinex      = "https://docs.bitfinex.com/docs/changelog"
+	pathBitmex        = "https://www.bitmex.com/static/md/en-US/apiChangelog"
+	pathANX           = "https://anxv3.docs.apiary.io/"
+	pathPoloniex      = "https://docs.poloniex.com/#changelog"
+	pathIbBit         = "https://api.itbit.com/docs"
+	pathBTCMarkets    = "https://api.btcmarkets.net/openapi/info/index.yaml"
+	pathEXMO          = "https://exmo.com/en/api/"
+	pathBitstamp      = "https://www.bitstamp.net/api/"
+	pathHitBTC        = "https://api.hitbtc.com/"
+	pathBitflyer      = "https://lightning.bitflyer.com/docs?lang=en"
+	pathLakeBTC       = "https://www.lakebtc.com/s/api_v2"
+	pathKraken        = "https://www.kraken.com/features/api"
+	pathAlphaPoint    = "https://alphapoint.github.io/slate/#introduction"
+	pathGetAllLists   = "https://api.trello.com/1/boards/%s/lists?cards=none&card_fields=all&filter=open&fields=all&key=%s&token=%s"
+	pathNewCard       = "https://api.trello.com/1/cards?%s&key=%s&token=%s"
+	pathChecklists    = "https://api.trello.com/1/checklists/%s/checkItems?%s&key=%s&token=%s"
+	apiKey            = ""
+	apiToken          = ""
+	updateChecklistID = "5dfc5a5377835d0ba025787a"
+)
+
+var verbose bool
+
+func main() {
+	flag.BoolVar(&verbose, "verbose", false, "Increases logging verbosity for API Update Checker")
+	flag.Parse()
+	updates, err := CheckUpdates(jsonFile)
+	if err != nil {
+		log.Println(err)
+	}
+	log.Println(updates)
+}
+
+// GetSha gets the sha of the latest commit
+func GetSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	finalPath := fmt.Sprintf(githubPath, repoPath)
+	if verbose {
+		log.Println(finalPath)
+	}
+	return resp, common.SendHTTPGetRequest(finalPath, true, false, &resp)
+}
+
+// CheckExistingExchanges checks if the given exchange exists
+func CheckExistingExchanges(fileName, exchName string) ([]ExchangeInfo, bool, error) {
```

Comment:
It would also be nice to output which exchanges are missing an API check. You can use exchange.Exchanges to find out which ones we support vs which ones we don't